automation/src/test/java/apiproject/apiTests.java

```diff
@@ -0,0 +1,104 @@
+package apiproject;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import static io.restassured.RestAssured.*;
+import static org.hamcrest.Matchers.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.json.simple.JSONObject;
+
+public class apiTests {
+	@Test
+	public void canVerifyEmailAddress() {
+		get("/api/users/2").then().statusCode(200).assertThat().body("data.email", equalTo("janet.weaver@reqres.in"));
+	}
+
+	@Test
+	public void canVerifyTokenValue() {
+		String tokenValue = "QpwL5tke4Pnpja7X4";
+		Map<String, String> requestMap = new HashMap<String, String>();
+		requestMap.put("email", "eve.holt@reqres.in");
+		requestMap.put("password", "cityslicka");
+		JSONObject request = new JSONObject(requestMap);
+
+		given().body(request.toJSONString()).post("/api/login").then().statusCode(200).body("token", equalTo(tokenValue));
+	}
+
+	@Test
+	public void canVerifyReponseCode204() {
+		// No Content success status - status code 204
+		Response response = delete("/api/users/2");
+		Assert.assertEquals(204, response.getStatusCode());
+	}
+
+	@Test
+	public void canUpdatePatch() {
+		Map<String, String> requestMap = new HashMap<String, String>();
+		requestMap.put("name", "morpheus2");
+		requestMap.put("job", "zion resident");
+		JSONObject request = new JSONObject(requestMap);
+
+		String timestamp = given().body(request.toJSONString()).when().patch("/api/users/2").then().extract()
+				.path("updatedAt");
+
+		Assert.assertNotNull(timestamp);
+	}
+
+	@Test
+	public void canUpdatePut() {
+		Map<String, String> requestMap = new HashMap<String, String>();
+		requestMap.put("name", "morpheus2");
+		requestMap.put("job", "zion resident");
+
+		JSONObject request = new JSONObject(requestMap);
+
+		String timestamp = given().body(request.toJSONString()).when().put("/api/users/2").then().extract()
+				.path("updatedAt");
+
+		System.out.println(
+				given().body(request.toJSONString()).when().put("/api/users/2").then().extract().asPrettyString());
+
+		Assert.assertNotNull(timestamp);
+	}
+	
+	@Test
+	public <T> void canDeserializeSingleUser() {
+		List<Map<String, T>> singleUser = get("/api/users/2").jsonPath().getList("data");
+		User<T> user = new User<T>(singleUser.get(0).get("id"), singleUser.get(0).get("email"),
```

Comment:
An easier way to deserialize is like the following:
List<Artwork> returnedArtworks = Arrays.asList(response.getBody().as(Artwork[].class));